pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -564,26 +571,30 @@ private void createPersistentTopic(final String topic, CompletableFuture<Topic>
             return;
         }
 
-        getManagedLedgerConfig(destinationName).thenAccept(config -> {
+        getTopicConfiguration(destinationName).thenAccept(config -> {
             // Once we have the configuration, we can proceed with the async open operation
 
-            managedLedgerFactory.asyncOpen(destinationName.getPersistenceNamingEncoding(), config,
+            managedLedgerFactory.asyncOpen(destinationName.getPersistenceNamingEncoding(), config.getManagedLedgerConfig(),
                     new OpenLedgerCallback() {
                         @Override
                         public void openLedgerComplete(ManagedLedger ledger, Object ctx) {
-                            PersistentTopic persistentTopic = new PersistentTopic(topic, ledger, BrokerService.this);
+                            PersistentTopic persistentTopic = new PersistentTopic(topic, ledger, BrokerService.this,
+                                    config);
```

Comment:
Before this PR, we were getting the per-topic (actually it's per-namespace) configuration only for ManagedLedger purposes: https://github.com/apache/incubator-pulsar/blob/aa40ebbe02f9b4f60b7f3910c293c50c18f35687/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java#L567

For deduplication, when creating the topic I need to know wether it's enabled or not, depending on the broker default and on the per-namespace policy.

`PersistentTopicConfiguration` is grouping both the policy and the customized `ManagedLedgerConfig`.

Reply:
> For deduplication, when creating the topic I need to know wether it's enabled or not
PersistentTopicConfiguration is grouping both the policy and the customized ManagedLedgerConfig.

Yes, I understood that part. But correct if I am wrong:
[MessageDeduplication](https://github.com/merlimat/pulsar/blob/4e1e522725b8c3707df71d7443bce3f58761d3d1/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/MessageDeduplication.java?utf8=%E2%9C%93#L335) checks namespace-dedup by directly fetching from `policies-cache` and  `PersistentTopicConfiguration.namespacePolicies` is not being used anywhere.  
so, I don't see any difference if we just return `ManagedLedgerConfig` instead `PersistentTopicConfiguration` because it seems we are not using it anywhere?

Reply:
Uhm, I guess you're right :) 

I have indeed changed the approach after adding the `PersistentTopicConfiguration` so that `isDeduplicationEnabled()` is checking the policy and returning a future. 

I'll remove `PersistentTopicConfiguration`

Reply:
> I'll remove `PersistentTopicConfiguration`

Should we revert `getTopicConfiguration()` to `getManagedLedgerConfig()` if we are not using `PersistentTopicConfiguration`?

Reply:
Ouch, I finally forgot to remove it. Will do that